### PR TITLE
Add new navItem() and navSpacer()

### PR DESCRIPTION
**Pull request by cpsievert**
This PR adds 2 new convenience functions for navs/tabsets:

* `navItem()`: for placing links, buttons, and inline forms within `navbarPage()`, `tabsetPanel()`, or `navlistPanel()`.
* `navSpacer()`: for adding space between navigation items (useful for creating right aligned `navItem()`)

It also adds an `align` argument to `navbarMenu()` for aligning the dropdown menu (useful now that it's easy to right-align navigation items):

```r
navItems <- list(
  tabPanel("a", "tab a"),
  tabPanel("b", "tab b"),
  navItem(
    tags$a(icon("github"), "Shiny", href = "https://github.com/rstudio/shiny", target = "_blank")
  ),
  navSpacer(),
  navbarMenu(
    "Other links", align = "right",
    tabPanel("b", "tab b"),
    navItem(
      tags$a(icon("r-project"), "RStudio", href = "https://rstudio.com", target = "_blank")
    )
  )
)

shinyApp(
  navbarPage(
    title = "Title",
    theme = bslib::bs_theme("navbar-bg" = "#0062cc"),
    !!!navItems
  ),
  function(...) {}
) 
```

Screen Shot 2021-03-23 at 3 03 17 PM

Also works with `tabsetPanel()`:

```r
shinyApp(
  fluidPage(
    theme = bslib::bs_theme(),
    tabsetPanel(
      !!!navItems, type = "pills"
    )
  ),
  function(...) {}
) 
```

Screen Shot 2021-03-23 at 3 31 52 PM

And `navlistPanel()` (in this context, `navSpacer()` functions as a vertical spacer):

```r
shinyApp(
  fluidPage(
    theme = bslib::bs_theme(),
    tags$head(tags$style(".tab-content * {height: 100vh}")),
    navlistPanel(!!!navItems)
  ),
  function(...) {}
) 
```

Screen Shot 2021-03-23 at 3 58 13 PM

### TODO

- [ ] Tests
- [ ] Documentation
- [ ] What should we do for the BS3 version of https://github.com/rstudio/bslib/pull/292?

### CODE REVIEW

File: R/bootstrap.R [hunk 1]
```diff
@@ -985,6 +1109,18 @@ buildTabItem <- function(index, tabsetId, foundSelected, tabs = NULL,
...
+  if (isNavLink(divTag)) {
+    return(
+      list(liTag = tags$li(class = "nav-item", divTag), divTag = NULL)
+    )
+  }
+
+  if (isNavForm(divTag)) {
+    return(
+      list(liTag = divTag, divTag = NULL)
+    )
```
Comment (cpsievert) [inline R/bootstrap.R:1121]: Is it a problem that this (might) be stuffed in a `<ul>` container

File: R/bootstrap.R [hunk 2]
```diff
@@ -687,6 +774,43 @@ tabPanelBody <- function(value, ..., icon = NULL) {
...
+  inherits(x, "shiny_nav_link")
+}
+
+#' @rdname tabPanel
+#' @export
+navForm <- function(...) {
+  form <- tags$form(class = "form-inline", ...)
+  class(form) <- c("shiny_nav_form", class(form))
+  form
+}
```
Comment (cpsievert) [inline R/bootstrap.R:803]: Could we combine these into a single function (navItem())?

File: inst/www/shared/navbarPage/navbarPage.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@if mixin-exists("media-breakpoint-up") {
+  @include media-breakpoint-up(sm) {
+    @include navbar-nav();
+  }
+} @else {
+  /* BS3 default navbar breakpoint */
+  @media (min-width: 768px) {
+    @include navbar-nav();
+  }
+}
```
Comment (cpsievert) [inline inst/www/shared/navbarPage/navbarPage.scss:32]: Maybe we should just roll this into shiny.scss?

**Comment by cpsievert** [comment]
Instead of adding `right`, maybe it'd be better to have a `tabSpacer()`:

```r
navbarPage(
  "Title", 
  tabPanel("a", "tab a"),
  tabPanel("b", "tab b"),
  tabSpacer(),
  navLink(icon("github"), "http://github.com/rstudio/shiny"),
  navForm(
    textInput("search", NULL, placeholder = "Search", width = "auto")
  ),
  theme = bslib::bs_theme("navbar-bg" = "#0062cc"),
  collapsible = TRUE
)
```

File: R/bootstrap.R (hunk 2 above)
Comment (wch) [reply R/bootstrap.R:803]: If we have `navItem()`, then links could be done with:

```R
navItem(tags$a(href = "...", icon(...), "Label"))
```

Also, to be correct HTML, each `navItem` would have to be an `<li>`.

File: inst/www/shared/shiny_scss/shiny.scss [hunk 4]
```diff
@@ -440,3 +440,45 @@ pre.shiny-text-output {
   /* override anything bootstrap sets for `.nav` */
   display: none !important;
 }
+
```
Comment (cpsievert) [inline inst/www/shared/shiny_scss/shiny.scss:443]: If we put this CSS in shiny.css, then this stuff won't work when statically rendered. Maybe this should come in as an HTML dependency on `navSpacer()`?

File: R/bootstrap.R [hunk 5]
```diff
@@ -690,6 +735,29 @@ tabPanelBody <- function(value, ..., icon = NULL) {
   tabPanel(title = NULL, ..., value = value, icon = icon)
 }
 
+#' @rdname tabPanel
+#' @export
+navItem <- function(...) {
+  item <- tags$li(class = "nav-item form-inline", ...)
+  class(item) <- c("shiny_nav_item", class(item))
+  item
```
Comment (cpsievert) [inline R/bootstrap.R:743]: `.shiny-nav-itme` is only used to identify, but could leverage shiny at some point

```suggestion
  tags$li(class = "shiny-nav-item nav-item form-inline", ...)
```

Comment (cpsievert) [reply R/bootstrap.R:743]: Also beware that BS5 dropped `.form-inline` https://getbootstrap.com/docs/5.0/migration/#forms-2

**Comment by cpsievert** [comment]
Superceded by https://github.com/rstudio/bslib/pull/319

**Pull request closed without merging**